utils/results/common.go

```diff
@@ -0,0 +1,612 @@
+	if multipleRoots {
+		return AppendUniqueImpactPathsForMultipleRoots(target, source)
+	}
+	impactPathMap := make(map[string][]services.ImpactPathNode)
+	for _, path := range target {
+		// The first node component id is the key and the value is the whole path
+		key := getImpactPathKey(path)
+		impactPathMap[key] = path
+	}
+
+	for _, path := range source {
+		key := getImpactPathKey(path)
+		if _, exists := impactPathMap[key]; !exists {
+			impactPathMap[key] = path
+			target = append(target, path)
+		}
+	}
+	return target
+}
+
+// getImpactPathKey return a key that is used as a key to identify and deduplicate impact paths.
+// If an impact path length is equal to directDependencyPathLength, then the direct dependency is the key, and it's in the directDependencyIndex place.
+func getImpactPathKey(path []services.ImpactPathNode) string {
+	key := path[RootIndex].ComponentId
+	if len(path) == DirectDependencyPathLength {
+		key = path[DirectDependencyIndex].ComponentId
+	}
+	return key
+}
+
+func SplitScaScanResults(results *SecurityCommandResults) ([]services.Violation, []services.Vulnerability, []services.License) {
+	var violations []services.Violation
+	var vulnerabilities []services.Vulnerability
+	var licenses []services.License
+	for _, scanTarget := range results.Targets {
+		for _, scaScan := range scanTarget.ScaResults.XrayResults {
+			violations = append(violations, scaScan.Violations...)
+			vulnerabilities = append(vulnerabilities, scaScan.Vulnerabilities...)
+			licenses = append(licenses, scaScan.Licenses...)
+		}
+	}
+	return violations, vulnerabilities, licenses
+}
+
+func GetCveApplicabilityField(cveId string, applicabilityScanResults []*sarif.Run, components map[string]services.Component) *formats.Applicability {
+	if len(applicabilityScanResults) == 0 {
+		return nil
+	}
+	applicability := formats.Applicability{}
+	resultFound := false
+	var applicabilityStatuses []jasutils.ApplicabilityStatus
+	for _, applicabilityRun := range applicabilityScanResults {
+		if rule, _ := applicabilityRun.GetRuleById(jasutils.CveToApplicabilityRuleId(cveId)); rule != nil {
+			applicability.ScannerDescription = sarifutils.GetRuleFullDescription(rule)
+			status := getApplicabilityStatusFromRule(rule)
+			if status != "" {
+				applicabilityStatuses = append(applicabilityStatuses, status)
+			}
+		}
+		result, _ := applicabilityRun.GetResultByRuleId(jasutils.CveToApplicabilityRuleId(cveId))
+		if result == nil {
+			continue
+		}
+		resultFound = true
+		// Add new evidences from locations
+		for _, location := range result.Locations {
+			fileName := sarifutils.GetRelativeLocationFileName(location, applicabilityRun.Invocations)
+			// TODO: maybe, move this logic to the convertor
+			if shouldDisqualifyEvidence(components, fileName) {
+				continue
+			}
+			applicability.Evidence = append(applicability.Evidence, formats.Evidence{
+				Location: formats.Location{
```

Comment:
Not crucial, more of a styling suggestion here - having a function that gets one type of location and return another - instead of the multiple similar functions.